botocove/cove_runner.py

```diff
@@ -33,19 +38,31 @@ def __init__(
         self.thread_workers = thread_workers
 
     def run_cove_function(self) -> CoveFunctionOutput:
-        # Run decorated func with all valid sessions
-        with ThreadPoolExecutor(max_workers=self.thread_workers) as executor:
-            jobs = [executor.submit(self.cove_thread, s) for s in self.sessions]
-            outputs = (f.result() for f in as_completed(jobs))
+        def submit_decorated_func_with_all_sessions(
```

Comment:
Think this can just be inline - a type hint on the list comp has the same impact without needing a seperate func

`futures: List["Future[CoveSessionInformation]"] = [executor.submit(self.cove_thread, s) for s in self.sessions]`

Reply:
I prefer functions with descriptive names over inline comments, so I had replaced your inline comment with a function named to similar effect. In this case the inline function doesn't do much for the readability of the code, so I'm okay with inlining it.

The original comment is still gone because I think the new comment about the pattern does a better job of explaining the code.

Mypy knows the type without a type hint, so I'd prefer to omit the hint. Do you have another reason for including it?

Reply:
I'm not particularly worried about that comment - the original that's removed isn't super useful and should probably be a docstring if anything. My comment was intended for the inner function being better served by just resolving the list comprehension in the outer func, or in this case, didn't feel it needed a func at all since the value it was adding in my eyes was the return annotation :)

> Mypy knows the type without a type hint, so I'd prefer to omit the hint. Do you have another reason for including it?

The reason I like that approach is for two reasons - annotations can be more useful for humans than just mypy inference - in this case it's mainly just an annotation saying "here's what this magic results in", and declaring the var type means that if mypy infers different it'll fail which is a useful signal for the occasional corner case.

Reply:
Agreed. I've added the type hint for clarify. How does it look now?